docs/reference/hooks/interchain-gas.mdx

```diff
@@ -31,28 +29,85 @@ struct StandardHookMetadata {
 }
 ```
 
-### Examples
+</TabItem>
+</Tabs>
+
+The `StandardHookMetadata` struct defines the fields required for metadata encoding:
+
+- `variant`: Specifies the metadata format version.
+- `msgValue`: Amount of native token sent with the message.
+- `gasLimit`: Gas limit for the `handle` function on the destination chain. Ensure this matches your simulation results.
+- `refundAddress`: Address to refund unused gas payments.
+
+To encode this metadata, use the `StandardHookMetadata.formatMetadata` library function. Direct encoding of the struct with `abi.encodePacked` is not supported in Solidity.
+
+#### Example Usage
+
+<Tabs groupId="lang">
+<TabItem value="sol" label="Solidity">
 
 ```solidity
-  bytes memory hookMetadata = StandardHookMetadata.formatMetadata(
-    0,                   // ETH message value
-    200000,              // 200k gas limit
-    address(this),       // refund address
-    bytes("")            // custom metadata
-  });
+// Example: Encoding metadata using StandardHookMetadata
+bytes memory metadata = StandardHookMetadata.formatMetadata(
+    0,               // ETH message value
+    200000,          // Custom gas limit
+    address(this),   // Refund address
+    bytes("")        // Optional custom metadata
+);
 ```
 
+### Determine and Override the Gas Limit
+
+1. Simulate and Benchmark Gas Usage:
+
+- Use tools like Tenderly or Foundry to simulate your transaction and measure gas consumption.
+- If gas usage exceeds `50,000` gas, calculate an appropriate `gasLimit` and update your metadata.
```

Comment:
this should specifically be simulate `handle` on your message recipient where the `from` address is the `mailbox` on that chain
we have a call to action for this in the explorer https://github.com/hyperlane-xyz/hyperlane-explorer/blob/03634076049bfce1611b4e41d1aa81910eab2962/src/features/messages/cards/TransactionCard.tsx#L326-L333